homeassistant/components/ebusd/__init__.py

```diff
@@ -0,0 +1,118 @@
+"""
+Support for Ebusd daemon for communication with eBUS heating systems.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/ebus/
+"""
+
+from datetime import timedelta
+import logging
+import socket
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_NAME, CONF_HOST, CONF_PORT, CONF_MONITORED_CONDITIONS)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+from homeassistant.util import Throttle
+
+from .const import (DOMAIN, SENSOR_TYPES)
+
+REQUIREMENTS = ['ebusdpy==0.0.16']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'ebusd'
+DEFAULT_PORT = 8888
+CONF_CIRCUIT = 'circuit'
+CACHE_TTL = 900
+SERVICE_EBUSD_WRITE = 'ebusd_write'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=15)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_CIRCUIT): cv.string,
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_MONITORED_CONDITIONS, default=[]): vol.All(
+            cv.ensure_list, [vol.In(SENSOR_TYPES['700'])])
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the eBusd component."""
+    conf = config[DOMAIN]
+    name = conf[CONF_NAME]
+    circuit = conf[CONF_CIRCUIT]
+    monitored_conditions = conf.get(CONF_MONITORED_CONDITIONS)
+    server_address = (
+        conf.get(CONF_HOST), conf.get(CONF_PORT))
+
+    try:
+        _LOGGER.debug("Ebusd component setup started.")
+        import ebusdpy
+        ebusdpy.init(server_address)
+        hass.data[DOMAIN] = EbusdData(server_address, circuit)
+
+        sensor_config = {
+            'monitored_conditions': monitored_conditions,
+            'client_name': name,
+            'sensor_types': SENSOR_TYPES[circuit]
+        }
+        load_platform(hass, 'sensor', DOMAIN, sensor_config, config)
+
+        hass.services.register(
+            DOMAIN, SERVICE_EBUSD_WRITE, hass.data[DOMAIN].write)
```

Comment:
We should include a service voluptuous validation schema when registering the service, since it takes parameters in the call data.